tests/components/statistics/test_sensor.py

```diff
@@ -1286,12 +1287,14 @@ async def test_initialize_from_database(
     assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == UnitOfTemperature.CELSIUS
 
 
+@pytest.mark.freeze_time(
```

Comment:
This test would freeze time too late and relied on the extra time between calling loop.time and the execution of the time.monotonic call.

By freezing time before the recorder is setup we don't have a loop in the recorder where it does the periodic tasks forever because the setup of the recorder was in a different time than the mocked time during async_wait_recording_done